lib/vanagon/project.rb

```diff
@@ -476,5 +477,36 @@ def generate_package
     def generate_packaging_artifacts(workdir)
       @platform.generate_packaging_artifacts(workdir, @name, binding)
     end
+
+    # Generate a json file which lists all of the dependant components of the project
+    def generate_dependencies_info
+      project_components = @components.map do |component|
+        components_hash = component.options
+        [:version, :url].each do |key|
+          if val = component.send(key)
+            components_hash[key] = val
+          end
+        end
+        [
+          component.name,
+          components_hash,
+        ]
+      end.to_h
+    end
+
+    def generate_manifest_json
+      build_time = Time.now
```

Comment:
This should be moved way out into top most scope, shouldn't it? It'd almost make sense as a global that can be referenced anywhere in the codebase. It's a handy piece of data to have access to.